src/runners.ts

```diff
@@ -35,7 +34,7 @@ export const runSetup = (params: {
       )
     }
 
-    merge(params.currentContext, contributedContext ?? {})
+    Object.assign(params.currentContext, contributedContext ?? {})
```

Comment:
This will do a shallow merge, we should actually add tests and docs about how merging works.

We definitely want a deep merge.

But maybe we do have something not desirable about the lodash algo. Can you investigate an isolated example of how the merge is not doing what you want/expect?

Reply:
I would expect it to simply blow up if the keys collide as there's no good way of approaching it - both merging and overriding have drawbacks - I'd treat it as an implementation error.

In [tRPC](https://trpc.io) I have a bunch of similar cases and the way I deal with it is that I make the server explode on bootup if you accidentally achieve "impossible" states (duplicate queries/multiple data transformers/[..]).

Reply:
If you don't actually utilize the behavior of being able to deep merge props in any of your apps already, I think you should just delete it. Always great to opt for keeping the API surface area slim & keeping deps to a minimum when making libs.